src/components/UserSelector.tsx

```diff
@@ -1,33 +1,77 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  onSetUser: (user: User) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({ users, onSetUser }) => {
+  const [dropdown, setDropdown] = useState(false);
+  const [userName, setUserName] = useState('Choose a user');
```

Comment:
The initial value should be null if there is no user. And add a conditional rendering instead 
```suggestion
  const [userName, setUserName] = useState(null);
```

Reply:
I have initial value user === null in app.tsx
Here initial value 'Choose a user', and I don't need conditional rendering
What is the meaning of this action?

Reply:
just logically that user name can't be `choose a user`
Here can be name or nothing, in our case `null`
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/55200530/8f31ca3f-05a8-4a52-ba36-79098dd18101)